src/xdg.c

```diff
@@ -212,44 +234,38 @@ xdg_toplevel_view_move(struct view *view, int x, int y)
 static void
 xdg_toplevel_view_close(struct view *view)
 {
-	wlr_xdg_toplevel_send_close(view->xdg_surface->toplevel);
+	wlr_xdg_toplevel_send_close(get_xdg_toplevel(view));
 }
 
 static void
 xdg_toplevel_view_maximize(struct view *view, bool maximized)
 {
-	wlr_xdg_toplevel_set_maximized(view->xdg_surface->toplevel, maximized);
+	wlr_xdg_toplevel_set_maximized(get_xdg_toplevel(view), maximized);
 }
 
 static void
 xdg_toplevel_view_set_activated(struct view *view, bool activated)
 {
-	struct wlr_xdg_surface *surface = view->xdg_surface;
-	if (surface->role == WLR_XDG_SURFACE_ROLE_TOPLEVEL) {
-		wlr_xdg_toplevel_set_activated(surface->toplevel, activated);
-	}
+	wlr_xdg_toplevel_set_activated(get_xdg_toplevel(view), activated);
```

Comment:
We're changing the logic slightly here. I can't think that it'll be a problem, but have not tested.

Reply:
The `wlr_xdg_surface::role` should never be anything other than `WLR_XDG_SURFACE_ROLE_TOPLEVEL` or we would not have created an `xdg_toplevel_view` in the first case. So I believe it is safe to remove the `if` here.

Reply:
Yes, let's go for it. I can't remember why it's there, but have tried running with a couple of GTK (xdg-shell) applications and it seems to behave as expected.